packages/react-aria-components/stories/Table.stories.tsx

```diff
@@ -393,6 +408,271 @@ const MyCheckbox = ({children, ...props}: CheckboxProps) => {
+          <Column isRowHeader={column.isRowHeader}>{column.name}</Column>
+        )}
+      </TableHeader>
+      <TableBody items={rows} dependencies={[args.isLoadingMore]}>
+        {(item) => (
+          <MyRow columns={columns} isLoadingMore={item.id === 4 && args.isLoadingMore}>
+            {(column) => {
+              return <Cell>{item[column.id]}</Cell>;
+            }}
+          </MyRow>
+        )}
+      </TableBody>
+    </Table>
+  );
+};
+
+export const TableLoadingRowRenderWrapperStory = {
+  render: TableLoadingRowRenderWrapper,
+  args: {
+    isLoadingMore: false
+  },
+  name: 'Table loading, row renderer wrapper and dep array'
+};
+
+
+function renderEmptyLoader({isLoading, tableWidth = 400}) {
+  let contents = isLoading ? 'Loading spinner' : 'No results found';
+  return <div style={{height: 'inherit', position: 'sticky', top: 0, left: 0, width: tableWidth}}>{contents}</div>;
+}
+
+const RenderEmptyState = (args: {isLoading: boolean}) => {
+  let {isLoading} = args;
+  return (
+    <Table aria-label="Files" selectionMode="multiple">
+      <TableHeader columns={columns}>
+        <Column><MyCheckbox slot="selection" /></Column>
+        <Collection items={columns}>
+          {(column) => (
+
+            <Column isRowHeader={column.isRowHeader}>{column.name}</Column>
+          )}
+        </Collection>
+      </TableHeader>
+      <TableBody renderEmptyState={() => renderEmptyLoader({isLoading})}>
+        <Collection items={[]}>
+          {(item) => (
+            <Row columns={columns}>
+              {(column) => {
+                return <Cell>{item[column.id]}</Cell>;
+              }}
+            </Row>
+          )}
+        </Collection>
+      </TableBody>
+    </Table>
+  );
+};
+
+export const RenderEmptyStateStory  = {
+  render: RenderEmptyState,
+  args: {
+    isLoading: false
+  },
+  name: 'Empty/Loading Table rendered with TableLoadingIndicator collection element'
+};
+
+interface Character {
+  name: string,
+  height: number,
+  mass: number,
+  birth_year: number
+}
+
+const OnLoadMoreTable = () => {
+  let list = useAsyncList<Character>({
+    async load({signal, cursor}) {
+      if (cursor) {
+        cursor = cursor.replace(/^http:\/\//i, 'https://');
+      }
+
+      // Slow down load so progress circle can appear
+      await new Promise(resolve => setTimeout(resolve, 4000));
+      let res = await fetch(cursor || 'https://swapi.py4e.com/api/people/?search=', {signal});
+      let json = await res.json();
+      return {
+        items: json.results,
+        cursor: json.next
+      };
+    }
+  });
+
+  let isLoading = list.loadingState === 'loading' || list.loadingState === 'loadingMore';
+  let scrollRef = useRef<HTMLDivElement>(null);
+  let memoedLoadMoreProps = useMemo(() => ({
+    isLoading: isLoading,
+    onLoadMore: list.loadMore
+  }), [isLoading, list.loadMore]);
+  useLoadMore(memoedLoadMoreProps, scrollRef);
+
+  return (
+    // TODO: but why typescript?
```

Comment:
I don't understand this TODO.

Reply:
ah thanks for the reminder, I can remove this